src/mlpack/tests/layer_names_test.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file layer_names_test.cpp
+ *
+ * Tests for testing the string representation of
+ * layers in mlpack's ANN module.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/layer_names.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace ann;
+
+BOOST_AUTO_TEST_SUITE(LayerNamesTest);
+
+/**
+ * Test if the LayerNameVisitor works properly.
+ */
+BOOST_AUTO_TEST_CASE(LayerNameVisitorTest)
+{
+  LayerTypes<> atrousConvolution = new AtrousConvolution<>();
+  LayerTypes<> alphaDropout = new AlphaDropout<>();
+  LayerTypes<> batchNorm = new BatchNorm<>();
+  LayerTypes<> constant = new Constant<>();
+  LayerTypes<> convolution = new Convolution<>();
+  LayerTypes<> dropConnect = new DropConnect<>();
+  LayerTypes<> dropout = new Dropout<>();
+  LayerTypes<> flexibleReLU = new FlexibleReLU<>();
+  LayerTypes<> layerNorm = new LayerNorm<>();
+  LayerTypes<> linear = new Linear<>();
+  LayerTypes<> linearNoBias = new LinearNoBias<>();
+  LayerTypes<> maxPooling = new MaxPooling<>();
+  LayerTypes<> meanPooling = new MeanPooling<>();
+  LayerTypes<> multiplyConstant = new MultiplyConstant<>();
+  LayerTypes<> reLULayer = new ReLULayer<>();
+  LayerTypes<> transposedConvolution = new TransposedConvolution<>();
+  LayerTypes<> identityLayer = new IdentityLayer<>();
+  LayerTypes<> tanHLayer = new TanHLayer<>();
+  LayerTypes<> eLU = new ELU<>();
+  LayerTypes<> hardTanH = new HardTanH<>();
+  LayerTypes<> leakyReLU = new LeakyReLU<>();
+  LayerTypes<> pReLU = new PReLU<>();
+  LayerTypes<> sigmoidLayer = new SigmoidLayer<>();
+  LayerTypes<> logSoftMax = new LogSoftMax<>();
+  LayerTypes<> unsupportedLayer = new BilinearInterpolation<>(); /* Bilinear
+  interpolation is not yet supported by the string converter */
+
+  BOOST_REQUIRE_EQUAL(boost::apply_visitor(LayerNameVisitor(),
```

Comment:
Haven't tested it but perhaps:
``BOOST_REQUIRE(boost::apply_visitor(LayerNameVisitor(), atrousConvolution) == "atrousconvolution");``

will solve the build issue.